File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.swift
import Foundation

var num  = 0
var answer = 0
var inputs: [String] = []
if let count = readLine(),
   let n = Int(count) {
    num = n
    for _ in 0..<n {
        if let line = readLine() {
            inputs.append(line)
        }
    }
}
var check = true
for input in inputs {
    check = true
    var saveWords: [String] = []
    let words = input
    for word in words {
        let checkWord = saveWords.last ?? ""
        if checkWord != String(word) {
            if saveWords.contains(String(word)) {
                check = false
            }
        }
        saveWords.append(String(word))
    }
    if check {
        answer += 1
    }
}
print(answer)

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
n = int(input())

answer = 0

for _ in range(n):
    word = input()
    seen = set()
    prev = ""

    flag = True

    for w in word:
        if w != prev:
            if w in seen:
                flag = False
                break
            seen.add(w)
        prev = w

    if flag:
        answer += 1

print(answer)
        
    